docs/dev/zksync-v2/aa.md

```diff
@@ -91,25 +159,93 @@ This is one of the whitelisted calls, where the account logic is allowed to call
 
 ### Sending transactions from an account
 
-For now, only EIP712 transactions are supported. To submit a transaction from a specific account, you should provide the `aaParams` object with the `from` and `signature` fields in the custom data:
+For now, only EIP712 transactions are supported. To submit a transaction from a specific account, you should provide `from` field of the transaction as the address of the sender and `customSignature` field of the `customData` with the signature for the account.
 
 ```ts
 import { utils } from "zksync-web3";
 
 // here the `tx` is a `TransactionRequest` object from `zksync-web3` SDK.
 // and the zksyncProvider is the `Provider` object from `zksync-web3` SDK connected to zkSync network.
+tx.from = aaAddress;
 tx.customData = {
   ...tx.customData,
-  aaParams: {
-    from: aaAddress,
-    signature: aaSignature,
-  },
+  customSignature: aaSignature,
 };
 const serializedTx = utils.serialize({ ...tx });
 
 const sentTx = await zksyncProvider.sendTransaction(serializedTx);
 ```
 
+## Paymasters
+
+Paymasters are accounts that can compensate for other accounts' transactions. Imagine being able to pay fees for users of your protocol! The other important use-case of paymasters is to facilitate paying fees in ERC20 tokens. While ETH is the main token of zkSync, paymasters can provide the ability to exchange ERC20s to ETH on the fly.
+
+If users want to interact with a paymaster, they should provide the non-zero `paymaster` address in their EIP712 transaction. The input data to the paymaster is provided in the `paymasterInput` field of the paymaster.
+
+### Paymaster verification rules
+
+
+::: warning Not implemented yet
+
+The verification rules are not fully enforced right now. Even if your paymaster works right now, it might stop working in the future if it does not follow the rules below.
+
+:::
+
+Since multiple users should be allowed to access the same paymasters, malicious paymasters *can* do a DoS attack on our system. To work around this, a system similar to the [EIP4337 reputation scoring](https://eips.ethereum.org/EIPS/eip-4337#reputation-scoring-and-throttlingbanning-for-paymasters) will be used.
+
+Unlike in the original EIP, paymasters are allowed to touch any storage slots. Also, the paymaster won't be throttled if either of the following is true:
+
+- More than `X` minutes have passed since the verification has passed on the API nodes. (The exact value of `X` is TBD).
```

Comment:
```suggestion
- More than `X` minutes have passed since the verification has passed on the API nodes. (The exact value of `X` will be defined later).
```